src/index.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ImprovedStreamingClient} from './helpers';
+import * as v1p1beta1 from './v1p1beta1';
+import * as v1 from './v1';
+
+// The following code is adapted from http://www.typescriptlang.org/docs/handbook/mixins.html
+// tslint:disable-next-line no-any
+Object.defineProperty(
+  v1.SpeechClient.prototype,
+  'streamingRecognize',
```

Comment:
What are the main benefits of the `ImprovedStreamingClient` mix-in? better types in the response?

I wonder if we should consider upstreaming some of this to the generator itself, worth making a tracking ticket?

Reply:
This looks like a speech-specific change. They want `streamingRecognize`, which is a bi-di streaming method, to accept a separate config object:

```
  streamingRecognize(
    streamingConfig?:
      | protosTypes.google.cloud.speech.v1.IStreamingRecognitionConfig
      | protosTypes.google.cloud.speech.v1p1beta1.IStreamingRecognitionConfig,
    options?: gax.CallOptions
  )
```

By our design, bi-directional streaming calls don't accept any request as a parameter (only as a stream). This one is unique.